hack/verify-api-groups.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# locate all API groups by their packages and versions
+
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+prefix="${KUBE_ROOT%"k8s.io/kubernetes"}"
+
+register_files=()
+while IFS= read -d $'\0' -r file ; do
+	register_files+=("${file}")
+done < <(find ${KUBE_ROOT}/pkg/apis -name register.go -print0)
+
+# every register file should contain a GroupName.  Gather the different representations.
+# 1. group directory name for go2idl client gen
+# 2. external group versions for init.sh all APIs list
+# 3. install packages for inclusion in import_known_versions files
+group_dirnames=()
+external_group_versions=()
+expected_install_packages=()
+for register_file in "${register_files[@]}"; do
+	package=${register_file#${prefix}}
```

Comment:
You went through all the effort in your `find` step to handle whitespace in file names, but it is lost if you don't quote things after that:
```sh
package="${register_file#"${prefix}"}"
        ^                ^         ^ ^
```